app/src/main/java/com/cupcake/chickenmasala/ui/fragment/home/adapter/HomeRecyclerAdapter.kt

```diff
@@ -0,0 +1,225 @@
+                ViewPagerViewHolder(view)
+            }
+            RECENT_FOOD -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_horizontal_recipes, parent, false)
+                RecentFoodViewHolder(view)
+            }
+            CHIPS_FILTER -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_chips_home, parent, false)
+                ChipsViewHolder(view)
+            }
+            HORIZONTAL_RECIPE -> {
+                val view = LayoutInflater.from(parent.context)
+                    .inflate(R.layout.item_vertical_recipes, parent, false)
+                FilteredFoodViewHolder(view)
+            }
+            else -> throw Exception(" UNKNOWN VIEW TYPE")
+        }
+    }
+
+    override fun onBindViewHolder(holder: BaseViewHolder, position: Int) {
+        when (holder) {
+            is ViewPagerViewHolder -> bindViewPager(holder, position)
+            is RecentFoodViewHolder -> bindRecentFood(holder, position)
+            is ChipsViewHolder -> bindChipsFiltered(holder, position)
+            is FilteredFoodViewHolder -> bindFilteredFood(holder, position)
+        }
+    }
+
+    private fun bindChipsFiltered(holder: ChipsViewHolder, position: Int) {
+        holder.binding.chipsFilter.setOnCheckedStateChangeListener { _, checkedIds ->
+            listener.onChipClicked(checkedIds[0])
+        }
+    }
+
+    private fun bindFilteredFood(holder: FilteredFoodViewHolder, position: Int) {
+        val recipe = items[position].item as Recipe
+        holder.binding.apply {
+            cuisineName.text = recipe.cuisine
+            recipeName.text = recipe.translatedRecipeName
+            recipeTime.text = recipe.totalTimeInMin.toString()
+            cardImage.setImage(recipe.imageUrl)
+            root.setOnClickListener {
+                listener.onCardClicked(recipe.id)
+            }
+        }
+    }
+
+    private fun bindRecentFood(holder: RecentFoodViewHolder, position: Int) {
+        val recentRecipes = items[position].item as List<Recipe>
+        val horizontalAdapter =
+            HorizontalRecipeRecyclerAdapter(listener)
+        horizontalAdapter.submitList(recentRecipes)
+        holder.binding.apply {
+            recyclerViewHorizontal.adapter = horizontalAdapter
+        }
+    }
+
+    private fun bindViewPager(holder: ViewPagerViewHolder, position: Int) {
+        val advices = items[position].item as List<HealthAdvice>
+        val viewPager = holder.binding.viewPager
+        val adapter = ViewPagerAdapter(viewPager, advices, listener)
+        holder.binding.apply {
+            viewPager.adapter = adapter
+            setupViewPager(viewPager, advices)
+            setUpTransformer(viewPager)
+            viewAll.setOnClickListener {
+                listener.onViewAllButtonClicked()
+            }
+
+        }
+    }
+
+    private fun ItemViewPagerBinding.setupViewPager(
```

Comment:
Don't use the private   extension  function  and don't use it inside the class 
the recommended uses of extension outside the class and call it in any class